docs/HANDLED_RULES.md

```diff
@@ -10,6 +10,7 @@ Sorald can currently repair violations of the following rules:
     * ["compareTo" should not return "Integer.MIN_VALUE"](#compareto-should-not-return-integermin_value-sonar-rule-2167) ([Sonar Rule 2167](https://rules.sonarsource.com/java/RSPEC-2167))
     * ["getClass" should not be used for synchronization](#getclass-should-not-be-used-for-synchronization-sonar-rule-3067) ([Sonar Rule 3067](https://rules.sonarsource.com/java/type/Bug/RSPEC-3067))
     * ["hashCode" and "toString" should not be called on array instances](#hashcode-and-tostring-should-not-be-called-on-array-instances-sonar-rule-2116) ([Sonar Rule 2116](https://rules.sonarsource.com/java/RSPEC-2116))
+    * ["equals(Object obj)" should test argument type](#equalsobject-obj-should-test-argument-type-sonar-rule-2097) ([Sonar Rule 2097](https://rules.sonarsource.com/java/RSPEC-2097))
```

Comment:
Yeah I was a bit confused by the order here, because the list is not sorted alphabetically (upper and lowercase do not matter for alphabetical sorting, `T` does not come before `e` for example)), nor is it sorted lexicographically (because then all rules starting with `"` would be in succession).

I think what was originally meant is lexicographical order, whereas @khaes-kth and I used alphabetical order, ignoring any non-alphabetical characters, which is why it's now not ordered in any way. I think alphabetical ordering is more intuitive than full-on lexicographical ordering, but for now I'll just restore the lexicographical ordering. Once I get around to #367 this will be a non-issue, this file is slightly too complex to edit manually imo.

Reply:
Ok then.

Reply:
Erm, I think you merged before I pushed the edited position xD